### PR TITLE
[on hold] Enable checkstyle on Utils and IntelliJ.

### PR DESCRIPTION
**Pull request by Eskibear**
- For Utils: Enable checkstyle in validate phase.
- For IntelliJ: See [ignoreFailures](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.quality.Checkstyle.html#org.gradle.api.plugins.quality.Checkstyle:ignoreFailures)
- For build script: Add checkstyleMain in one of the build command, so we can see the log in CI server.

### CODE REVIEW

**Comment by Eskibear** [comment]
Related issue: #1960

File: Utils/pom.xml [hunk 1]
```diff
@@ -22,8 +22,8 @@
   ~  */
   -->
 
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" 
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (wezhang) [inline Utils/pom.xml]: We can move this coding style change out as a standalone patch.

File: Utils/pom.xml [hunk 2]
```diff
@@ -84,20 +84,30 @@
...
+                    <dependency>
+                        <groupId>com.microsoft.azuretools</groupId>
+                        <artifactId>check-tools</artifactId>
+                        <version>1.0.0</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
+                        <artifactId>checkstyle</artifactId>
+                        <version>7.7</version>
+                    </dependency>
```
Comment (wezhang) [inline Utils/pom.xml]: The same as above.

File: Utils/pom.xml (hunk 1 above)
Comment (Eskibear) [reply Utils/pom.xml]: Sure.

File: Utils/pom.xml (hunk 2 above)
Comment (Eskibear) [reply Utils/pom.xml]: Here the indentation is wrong, but I'll follow your suggestion to fix it in another PR.

**Comment by wezhang** [comment]
weird error:

```
Could not resolve all files for configuration ':checkstyle'.
> Could not find commons-collections.jar (commons-collections:commons-collections:3.2.2).
  Searched in the following locations:
      file:/var/lib/jenkins/workspace/Azure_Toolkits_for_Java/AzureToolsManager_PR/.repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar
```

**Comment by Eskibear** [comment]
@wezhang 
Yes, investigating why the dependency package is not installed automatically. Do you have any thought?

BTW, it reminds me the merging methods discussion (squash vs rebase). Now I think squash might be better, because in code review we cannot avoid to submit more minor commits to fix something, and I think it would be more clear to squash them into one.

**Comment by wezhang** [comment]
> BTW, it reminds me the merging methods discussion (squash vs rebase). Now I think squash might be better, because in code review we cannot avoid to submit more minor commits to fix something, and I think it would be more clear to squash them into one.

`Squash` will erase the development intermediate progress. Specially, we have no development or version branches currently, the PR branch is used for some large changes feature. If all comments are squashed, we'll lost lot of ideas.

I prefer the `rebase` and the minimal patch size as small/atomic as possible with rich comments in codes and commits. And I also agree that the dev should squash the minor commits by himself (It's too ideal, I know 😃 ).

Just my 2 cents.

**Comment by wezhang** [comment]
> Yes, investigating why the dependency package is not installed automatically. Do you have any thought?

Looks like a Gradle plugin dependence missed, update Intellij Gradle plugin version?

**Comment by Eskibear** [comment]
I also encounter issues like `jetty-http.jar not found`. The dependency is introduced by htmlunit in hdinsght-node-common module, but the jar file is not downloaded, only the pom.xml is downloaded. 

Due to the lack of context information of the module, I think I'd rather follow @wezhang 's suggestion to enable the checkstyle module by module. 
First enable it in  azuretools-core and azure-explorer-common , see https://github.com/Microsoft/azure-tools-for-java/pull/1986

**Comment by Eskibear** [comment]
Close this, follow discussion of checkstyle rules in #1960

**Pull request closed without merging**